### PR TITLE
Dev instructions improvement

### LINKED ISSUE #14218: When developing Logstash, dependency gems are not installed.
### Tell us about the issue

**Description:** 
According to Logstash readme instruction, `rake bootstrap` should bootstrap the environment. However, when executing the command, resulted gems not found issue.
```
...
Could not find belzebuth-0.2.3, benchmark-ips-2.10.0, ci_reporter_rspec-1.0.0, flores-0.0.7, json-schema-2.8.1, logstash-devutils-2.4.0-java, rack-test-1.1.0, rspec-3.11.0, webmock-3.14.0, ci_reporter-2.0.0, fivemat-1.3.7, gem_publisher-1.5.0, kramdown-1.14.0, rspec-wait-0.0.9, rspec-core-3.11.0, rspec-expectations-3.11.0, rspec-mocks-3.11.1, crack-0.4.5, hashdiff-1.0.1, builder-3.2.4, rspec-support-3.11.0, diff-lcs-1.5.0 in any of the sources
Run `bundle install` to install missing gems
```

My investigations show, `rake bootstrap` is using a global bundle and installing gems globally. We have two options:
 - we have a ready gradle task to install gems locally where we can execute: `./gradlew installDevelopmentGems` (renamed recently)
 - OR set the local bundle and install dependencies:
```
bundle config set --local path vendor/bundle
bundle install
```

**URL:https://github.com/elastic/logstash/blob/main/README.md**

### PR DESCRIPTION
**Pull request by mashhurs**
## Release notes

[rn:skip]

## What does this PR do?

Add an instruction to resolve the failure when bootstrapping the LS environment.

## Why is it important/What is the impact to the user?
Users are not impacted. The engineer who want to contribute to LS environment might face dependency not installed issue.

## Checklist

~- [ ] My code follows the style guidelines of this project~
~- [ ] I have commented my code, particularly in hard-to-understand areas~
~- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)~
~- [ ] I have added tests that prove my fix is effective or that my feature works~
- [x] I have made corresponding changes to the documentation

## Author's Checklist

- [ ]

## Related issues

- Closes #14218

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -313,7 +313,7 @@ tasks.register("installDefaultGems") {
     }
 }
 
-tasks.register("installTestGems") {
+tasks.register("installDevelopmentGems") {
     dependsOn bootstrap
     doLast {
       rake(projectDir, buildDir, 'plugin:install-development-dependencies')
```
Comment (mashhurs) [inline build.gradle:319]: internally installing dependencies.

**Review by yaauie — COMMENTED** [review]
As a person who has bootstrapped Logstash a number of times over the years, I don't think I've done a stand-alone `rake bootstrap` or directly invoked `bundle` in 4+ years since we finished moving most things to gradle. 

I am wary about instructions that tell a newcomer "If you are using gradle" or "If you want to build with bundle" because even as a seasoned Logstash developer I generally don't care which tool I need to use as long as it gets the project bootstrapped. Telling a user how to build with bundle also implies that they have bundler installed which can be its own [can-of-worms](https://www.mentalfloss.com/article/31039/how-did-term-open-can-worms-originate).

***

After cloning (and frequently after changing major branches), my incantation to get everything magically working is `<ctrl>`+`r`, `./gra`, `<tab>`, which resolves to:

~~~
./gradlew clean bootstrap assemble installDefaultGems
~~~

These tasks, as I understand them:
 - `clean`: blow away build caches and stuff that may get in the way (not necessary on a fresh checkout)
 - `bootstrap`: install jruby and the dependencies required to run Logtash itself
 - `assemble`: compile the java bits and put them in the right places
 - `installDefaultGems`: install the ruby runtime dependencies that make Logstash _Logstash_, such as the default plugins.

While I don't _always_ need all parts of this sequence, typically invoking the whole sequence is less work than determining exactly what parts I need and ensures that the types of things that get out of sync when switching major branches are out of the way.

**Comment by mashhurs** [comment]
> As a person who has bootstrapped Logstash a number of times over the years, I don't think I've done a stand-alone `rake bootstrap` or directly invoked `bundle` in 4+ years since we finished moving most things to gradle.
> 
> I am wary about instructions that tell a newcomer "If you are using gradle" or "If you want to build with bundle" because even as a seasoned Logstash developer I generally don't care which tool I need to use as long as it gets the project bootstrapped. Telling a user how to build with bundle also implies that they have bundler installed which can be its own [can-of-worms](https://www.mentalfloss.com/article/31039/how-did-term-open-can-worms-originate).
> 
> After cloning (and frequently after changing major branches), my incantation to get everything magically working is `<ctrl>`+`r`, `./gra`, `<tab>`, which resolves to:
> 
> ```
> ./gradlew clean bootstrap assemble installDefaultGems
> ```
> 
> These tasks, as I understand them:
> 
> * `clean`: blow away build caches and stuff that may get in the way (not necessary on a fresh checkout)
> * `bootstrap`: install jruby and the dependencies required to run Logtash itself
> * `assemble`: compile the java bits and put them in the right places
> * `installDefaultGems`: install the ruby runtime dependencies that make Logstash _Logstash_, such as the default plugins.
> 
> While I don't _always_ need all parts of this sequence, typically invoking the whole sequence is less work than determining exactly what parts I need and ensures that the types of things that get out of sync when switching major branches are out of the way.

I also use clean assemble and no issue with building and running LS. However, one critic case I would say is when running the RSpec test cases. Eg., whenever I add new unit test I try to run it separately (with `bin/rspec /path/to/unit/test/spec_file.rb:line_number`) since entire test decreases productivity. This case seems requires bundle and if it is not a local one, execution ends up raising an exception that dependencies not found. And I do remember with previous JRuby versions (9.1.x) I hadn't issue with `rspec` but when we moved to `9.3.4.0` it is happening to me and to other contributors as well.

**Comment by roaksoax** [comment]
lgtm!

**Review by roaksoax — APPROVED** [review]

**Pull request merged**